otx/algorithms/detection/tasks/inference.py

```diff
@@ -267,6 +271,15 @@ def _init_test_data_cfg(self, dataset: DatasetEntity):
         )
         return data_cfg
 
+    def _update_stage_module(self, stage_module):
+        if hasattr(self._recipe_cfg, "train_type"):
```

Comment:
Same comments above. I think we don't need to check train_type inside recipe_cfg. Just checking it by if `self.train_type = TrainType.SEMISUPERVISED`